helm/storage.rst

```diff
@@ -0,0 +1,97 @@
+
+  pvc: What pvc is used to access this data.  If supplied the pvc will be mounted in each component that is required to access the data.  (The option is not specified for blob storage accessed via a url.)
+
+  secret: If a secret is required to access the storage (e.g. an access token), the name of the entry in the secrets section of the helm chart.  (Not the k8s secret name.)
+
+  numDevices: The number of logical storage devices.  An advanced option.  This allows data to be striped over multiple storage locations.  If it is set, then it assumes pvc defines a set of pvcs: ``pvc-1`` .. ``pvc-<n>``, which are mounted at locations ``prefix/d1`` .. ``prefix/d<n>``.  
+
+  replication: Not normally set, but if specified data will be copied in the background to each of the storage planes in the list.
+
+The planes are used by refering to them in the plane tag in the appropriate storage section.  E.g. ``--set storage.daliStorage.plane=dali-plane``.  If no plane is referenced for a storage class the simpler configuration will be used for that class.
+
+By default the different engines - thor, roxie, eclagent write their data to the default data storage plane.  This can be overridden by defining the plane key within the component.  The spill location can be set through the spillPlane option.  It is also possible to override the target storage plane by using the CLUSTER options in the ECL language.  This allows the ECL programmer to write data to different storage planes depending on the performance/cost requirements.  The engines can read data from any of the data planes.
+(Currently we use cluster to mean which set of machines to run a query on, and where the data is stored.  We probably need to separate those with a new attribute STORE(?) .)
+
+A plane can only be used for one of dali, data or dlls or spills.  An error will be reported if a plane is used for more than one purpose.  This prevents ECL queries having access to dali or dlls, (for reading or corruption).
+
+Bare-metal compatible storage planes are similar, and are generally derived from the existing deployment.  They'll be documented later.
+
+How the simple modes map onto the storage planes
+------------------------------------------------
+
+If no planes are mentioned for a specific class, then the following plane is implicitly created for the class::
+
+  name: hpcc-<class>-plane
+  prefix: /var/lib/HPCCSystems/hpcc<class>
+  pvc: storage.<class>Storage.pvc
+
+If no pvc was defined then an implicit pvc will be used.
+
+Notes
+=====
+I suggest replication is only support if explicitly defined using the storage planes, or if it is deduced from the group information on a bare metal system.
+
+In the future we may want to restrict which storage planes
+I suggest we deprecate the following options:
+global.defaultDataPath
+global.defaultMirrorPath
```

Comment:
I wonder if we can issue a warning that deprecated in the schema, suggesting a plane should be defined if non-defaults are used.

Reply:
I could add a warning entry in the notes.